core/frontend/src/tile/map/MapLayerSources.ts

```diff
@@ -57,7 +57,7 @@ export class MapLayerSource implements MapLayerProps {
   }
 
   public toLayerSettings(): MapLayerSettings | undefined {
-    const layerSettings = MapLayerSettings.fromJSON(this.toJSON());
+    const layerSettings = MapLayerSettings.fromJSON(this);
```

Comment:
@mdastous-bentley are there any tests for the map layer sources file? If not, mind adding some could help preventing a regression in the future.

Reply:
Please do.

Reply:
@pmconne  agreed, test added.